src/main/java/seedu/address/logic/commands/DeleteCommand.java

```diff
@@ -33,15 +37,40 @@ public DeleteCommand(Index targetIndex) {
     @Override
     public CommandResult execute(Model model) throws CommandException {
         requireNonNull(model);
-        List<Person> lastShownList = model.getFilteredPersonList();
 
-        if (targetIndex.getZeroBased() >= lastShownList.size()) {
+        ListType type = model.getCurrentListType();
+        List<Student> lastShownStudentList;
+        List<Tutor> lastShownTutorList;
+        List<TuitionClass> lastShownTuitionClassList;
+        String entityInformation;
+
+        try {
+            switch (type) {
+            case STUDENT_LIST:
+                lastShownStudentList = model.getFilteredStudentList();
+                Student studentToDelete = lastShownStudentList.get(targetIndex.getZeroBased());
+                model.deletePerson(studentToDelete);
+                entityInformation = studentToDelete.toString();
+                break;
+            case TUTOR_LIST:
+                lastShownTutorList = model.getFilteredTutorList();
+                Tutor tutorToDelete = lastShownTutorList.get(targetIndex.getZeroBased());
+                model.deletePerson(tutorToDelete);
+                entityInformation = tutorToDelete.toString();
+                break;
+            default:
```

Comment:
Would it be better to have a case for TUITIONCLASS_LIST and let the default throw an exception?

Reply:
Thats actually what I did last time maybe can discuss which is better later